frontend/src/pages/projects/screens/spawner/spawnerUtils.ts

```diff
@@ -141,6 +134,37 @@ export const getImageStreamDescription = (imageStream: ImageStreamKind): string
 export const getImageSteamOrder = (imageStream: ImageStreamKind): number =>
   parseInt(imageStream.metadata.annotations?.[IMAGE_ANNOTATIONS.IMAGE_ORDER] || '100');
 
+export const getCompatibleAcceleratorIdentifiers = (
+  object: ImageStreamKind | K8sDSGResource,
+): string[] => {
+  try {
+    const annotation = object.metadata.annotations?.['opendatahub.io/recommended-accelerators'];
+    // in the format of ["foo.com/gpu", "bar.com/gpu"]
+    if (annotation) {
+      const identifiers = JSON.parse(annotation);
+      if (Array.isArray(identifiers)) {
+        return identifiers;
+      }
+    }
+  } catch (error) {
+    // catch invalid json in metadata
```

Comment:
Are we silently failing here?

Reply:
yes... maybe i should throw an error but there seems no reason to. My reasoning is that this feature is for adding a recommended label to the respective accelerators and adding info/warnings. None of that is needed for a successful spawn. So if a cluster admin fails to use correctly formatted json. the user is blocked from using that specific image, bc of the thrown error.

Reply:
_raise UX issue_